scripts/publisherDVL1000.py

```diff
@@ -6,295 +6,354 @@
-		rospy.loginfo("Publishing sensor data from DVL Bottom-Track %s" % rospy.get_time())
-		pubBottom.publish(theDVL)
-		backupjson = getJson
-
-		theOdo.twist.twist.angular.z = unknown
-		global initZ
-		theOdo.pose.pose.position.z=-((BottomPressureData*10000)*10)/(997*9.81)
-		if (initZ == 0) and (abs(theDVL.velocity.x) < 10):
-			initZ = theOdo.pose.pose.position.z
-			rospy.logdebug(
-				"InitZ set to: %f" % (initZ)
-			)
-			
-		if (BottomXyzFom1Data != 10) and (BottomXyzFom2Data != 10) and (BottomXyzFomZbest != 10):
-			theOdo.twist.covariance = [BottomXyzFom1Data * BottomXyzFom1Data, unknown, unknown, unknown, unknown, unknown, unknown, BottomXyzFom2Data * BottomXyzFom2Data, unknown, unknown, unknown, unknown, unknown, unknown, BottomXyzFomZbest * BottomXyzFomZbest, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown]
-		pubOdo.publish(theOdo)
-
-
-        
-		#Pressure topic
-		thePressure.header.stamp = rospy.Time.now()
-		thePressure.header.frame_id = "dvl_link"
-		thePressure.fluid_pressure = BottomPressureData * 10000 #Convert dbar to Pascal
-		thePressure.variance = 30*30 #Should do a more accurate meassurement of the variance
-		pubPressure.publish(thePressure)
-	
+    global unknown
+    global backupjson
+    # Get JSON Data
+    getJson = cycleDVL()
+    try:
+        theData = json.loads(getJson)
+    except:
+        theData = json.loads(backupjson)
+
+    # Alternating data each time data is recieved. Each with unique ID(4123 and 8219) and dataset. Using IF to sort this out
+    # 8219 seem to have additional information like Status, Speed of Sound, Temperature
+    # More data if more modes of tracking is turned on. IDs 4125 and 8221 belongs to Water Tracking mode.
+    # IDs 4123 and 8219 belongs to Bottom Track mode.
+
     
-	#while not rospy.is_shutdown():
-		#rospy.loginfo("Publishing sensor data from DVL %s" % rospy.get_time())
-		#pub.publish(theDVL)
-		#rate.sleep()
-	rate.sleep()
-if __name__ == '__main__':
-	try:
-		rospy.init_node('DVL1000', anonymous=False)
-		rate = rospy.Rate(8) # 8hz
-  
-		websocket_address = rospy.get_param("/dvl/websocket_address", "ws://192.168.0.96:10100")
-  
-		#Websocket connection stuff. Make sure the websocket-client python library is installed.
-		ws = create_connection(websocket_address, subprotocols=["data-transfer-nortek"])
-		print("Connecting to DVL1000 via websocket...")
-		result = ws.recv()
-		print("Status: '%s'" % result)
-
-		backupjson = ''
-		unknown = 0 #The arbitrary value set for unknown values throughout the file
-		initZ = 0 #For resetting Z-value when initialized
-  
-		while not rospy.is_shutdown():
-			publishDVLdata()
-			
-	except rospy.ROSInterruptException:
-		pass
-        
-ws.close
+    # pubWater = rospy.Publisher('sensors/dvl/water', DVL, queue_size=10)
+
+    theDVL = DVL()
+    theDVLBeam = DVLBeam()
```

Comment:
Having "the" in front of the objects here has always seemed off to me, might just be personal taste, but it also doesn't really fit with the rest of our naming practices, soo idk.. Also peep the comment further below about creating the DVL() and DVLBeam() objects up here.

Reply:
I agree, the naming is weird - and so is the object creation style